dbms/src/Storages/Page/V3/spacemap/SpaceMapSTDMap.h

```diff
@@ -117,7 +117,17 @@ class STDMapSpaceMap
         {
             return end - start;
         }
-        return free_map.rbegin()->first;
+
+        const auto & last_node_it = free_map.rbegin();
+
+        // If last node is not [xxx, end]
+        // Then we should return `end` rather than last node start
+        if (last_node_it->first + last_node_it->second != end)
```

Comment:
Please explain why we need this check, and why this update solves https://github.com/pingcap/tiflash/issues/5076

Reply:
added